main.cpp

```diff
@@ -14,12 +14,23 @@ New/This/Pointers/References conclusion
          on the heap without leaking, without using smart pointers. 
  */
 
+struct A { };
 
+struct HeapA
+{
+    HeapA() 
+    {
+        pointerToA = new A;
+    }
```

Comment:
If you have to use a constructor that takes parameters, and those parameters initialize member variables, you should always prefer the member initialize list compared to initializing the variables inside the constructor body.
The reason is because the variables need to be instantiated/initialized when the class is created, and then when you assign to them in the ctor body, you are potentially re-initializing them.  that is a waste of resources.
- [ ] revise, use the member initializer list.